application/MainWindow.cpp

```diff
@@ -1563,95 +1491,40 @@ void MainWindow::on_actionLaunchInstanceOffline_triggered()
-	if (accounts->count() <= 0)
+	while (true)
 	{
-		// Tell the user they need to log in at least one account in order to play.
-		auto reply = CustomMessageBox::selectable(
-			this, tr("No Accounts"),
-			tr("In order to play Minecraft, you must have at least one Mojang or Minecraft "
-			   "account logged in to MultiMC."
-			   "Would you like to open the account manager to add an account now?"),
-			QMessageBox::Information, QMessageBox::Yes | QMessageBox::No)->exec();
+		MojangAuthSessionPtr session(new MojangAuthSession());
+		session->wants_online = online;
+		AuthTask *task = new AuthTask("minecraft", m_selectedInstance, session);
 
-		if (reply == QMessageBox::Yes)
+		ProgressDialog dlg(this);
+		if (online)
 		{
-			// Open the account manager.
-			on_actionManageAccounts_triggered();
+			dlg.setSkipButton(true, tr("Play Offline"));
 		}
-	}
-	else if (account.get() == nullptr)
-	{
-		// If no default account is set, ask the user which one to use.
-		AccountSelectDialog selectDialog(tr("Which account would you like to use?"),
-										 AccountSelectDialog::GlobalDefaultCheckbox, this);
-
-		selectDialog.exec();
-
-		// Launch the instance with the selected account.
-		account = selectDialog.selectedAccount();
-
-		// If the user said to use the account as default, do that.
-		if (selectDialog.useAsGlobalDefault() && account.get() != nullptr)
-			accounts->setActiveAccount(account->username());
-	}
-
-	// if no account is selected, we bail
-	if (!account.get())
-		return;
-
-	// we try empty password first :)
-	QString password;
-	// we loop until the user succeeds in logging in or gives up
-	bool tryagain = true;
-	// the failure. the default failure.
-	QString failReason = tr("Your account is currently not logged in. Please enter "
-							"your password to log in again.");
-
-	while (tryagain)
-	{
-		AuthSessionPtr session(new AuthSession());
-		session->wants_online = online;
-		auto task = account->login(session, password);
-		if (task)
+		dlg.exec(task);
+		if (dlg.exec(task) != QDialog::Accepted)
 		{
-			// We'll need to validate the access token to make sure the account
-			// is still logged in.
-			ProgressDialog progDialog(this);
-			if (online)
-				progDialog.setSkipButton(true, tr("Play Offline"));
-			progDialog.exec(task.get());
-			if (!task->successful())
-			{
-				failReason = task->failReason();
-			}
+			return;
 		}
+
 		switch (session->status)
 		{
-		case AuthSession::Undetermined:
+		case MojangAuthSession::Undetermined:
 		{
 			qCritical() << "Received undetermined session status during login. Bye.";
-			tryagain = false;
 			break;
 		}
-		case AuthSession::RequiresPassword:
+		case MojangAuthSession::RequiresPassword:
 		{
-			EditAccountDialog passDialog(failReason, this, EditAccountDialog::PasswordField);
-			if (passDialog.exec() == QDialog::Accepted)
-			{
-				password = passDialog.password();
-			}
-			else
-			{
-				tryagain = false;
-			}
+			qCritical() << "This should not be reachable";
```

Comment:
I think Github is derping something here.

```cpp
case MojangAuthSession::RequiresPassword:
{
	qCritical() << "This should not be reachable";
	break;
}
```

If a password is needed the task will fail, meaning ProgressDialog::exec returns QDialog::Rejected which returns on MainWindow.cpp#1512